src/components/KSlideout/KSlideout.vue

```diff
@@ -3,28 +3,50 @@
     <transition name="fade">
       <div
         v-if="isVisible"
-        class="panel-background"
-        @click="(event: any) => handleClose(event, true)"
+        :class="hasOverlay ? 'panel-background' : 'panel-background-transparent'"
       />
     </transition>
-    <transition
-      name="slide"
-    >
+    <transition name="slide">
       <div
         v-if="isVisible"
+        ref="slideOutRef"
         class="panel"
-        :class="{ isVisible: 'is-visible' }"
+        :class="{ 'is-visible': isVisible, 'border-styles': !hasOverlay }"
       >
-        <button
-          class="close-btn"
-          @click="(event: any) => handleClose(event, true)"
+        <!-- title slot -->
+        <div
+          v-if="title || hasTitle"
+          class="k-slideout-header-content"
         >
-          <KIcon
-            icon="close"
-            :size="KUI_ICON_SIZE_50"
-            view-box="0 0 24 24"
-          />
-        </button>
+          <slot name="title">
+            <p class="k-slideout-title">
+              {{ title }}
+            </p>
+          </slot>
+        </div>
+
+        <!-- header custom slot -->
+        <div v-if="hasHeader">
+          <slot name="header" />
+        </div>
+
```

Comment:
I don't think you want this to be two different slots? You don't mention the `title` slot in the docs and I would assume using the `header` slot would completely override display of the `title`/`title` slot.

Reply:
I will ping when its ready for review, I am still making changes.